src/libraries/System.Security.Cryptography.X509Certificates/tests/ExtensionsTests/SubjectAlternativeNameTests.cs

```diff
@@ -0,0 +1,202 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests.ExtensionsTests
+{
+    public static class SubjectAlternativeNameTests
```

Comment:
Some other test cases that I think are missing:

1. Using `CopyFrom` on an already hydrated instance. Something like:

    ```C#
    X509SubjectAlternativeNameExtension ext = new(someData);
    Assert.NotEmpty(ext.EnumerateDnsNames()); // Force decoding
    ext.CopyFrom(someOtherData);
    // Assert that ext has data from someOtherData, not someData
    ```